### ISSUE TITLE
No caching support

### ISSUE THREAD

**Issue by jslay88** [shauntarves/wyze-sdk#29]
To try and keep on the good side of Wyze, the sdk should implement caching on the api "get" calls (where states are not changing, etc), and add args to methods to override cache.

There are several methods for doing this (decorators, inline, etc), as well as existing modules/libraries from pypi that could be used.

**Comment by anthonystabile** [comment]
**I second this request!**  I implemented wyze-sdk as an interface in [Pytomation](http://www.pytomation.com/).  Most of the other Pytomation interfaces work with local hardware, instead of cloud based providers, and perform polling updates once every few seconds.  

As such, I was reluctant to code the update methods until I have reasonable throttles in place (i.e. check on a light status more frequently if the code believes there is occupancy in the room the light is in.)   Caching within the wyze-sdk would help a lot!

**Comment by shauntarves** [comment]
Hey @anthonystabile, I'm not sure that this library is the correct place to add caching support. In my opinion, this is "low-level" code to abstract the various API patterns/endpoints that Wyze uses into a common pattern of methods and models. It should be the next level up (the "application" layer) that adds support for caching and other performance-related functionality.

tl;dr; I don't plan on adding caching support, but would be happy to consider a pull request that adds it.
